pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -621,6 +630,28 @@ func makeHttpHealthCheck(name, path string, port int32) *compute.HttpHealthCheck
 	}
 }
 
+// applyHttpHealthCheckChanges applies user configurations to the new HC while
+// discard the changes that are not applicable to the HC. It returns whether
+// the HTTP HC is drifted and needs update.
+func applyHttpHealthCheckChanges(hc, newHC *compute.HttpHealthCheck) (drift bool) {
+	if hc.CheckIntervalSec > newHC.CheckIntervalSec {
+		newHC.CheckIntervalSec = hc.CheckIntervalSec
+	}
+	if hc.TimeoutSec > newHC.TimeoutSec {
+		newHC.TimeoutSec = hc.TimeoutSec
+	}
+	if hc.UnhealthyThreshold > newHC.UnhealthyThreshold {
+		newHC.UnhealthyThreshold = hc.UnhealthyThreshold
+	}
+	if hc.HealthyThreshold > newHC.HealthyThreshold {
+		newHC.HealthyThreshold = hc.HealthyThreshold
+	}
+	drift = hc.Port != newHC.Port || hc.RequestPath != newHC.RequestPath || hc.Description != newHC.Description
```

Comment:
remove this, also it is checking fields not set by the func, which is non-obvious

Reply:
Refactored to be a separate function.